src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -18,183 +29,856 @@ object HullmodExoticHandler {
+                // because the Optional must be present if we don't already have an entry in HullmodExoticHandler's map
+                // but if we do, we can just send whatever because it won't be used.
+                // Taking non-modular ships into account, we should check if we have an entry in the lookup map
+                // and if we do - we can send Optional.empty() since it won't be used
+                // but if we do not - we need to generate a list of our variants (which is most likely just this variant
+                // because the modules are handled first up there and they'd have already setup the whole entry
+                val variantListOptional = if (
+                        HullmodExoticHandler.doesEntryExist(
+                                hullmodExotic = hullmodExotic,
+                                fleetMember = member
+                        )
+                ) {
+                    // Entry exists, lets just send an empty optional
+                    Optional.empty()
+                } else {
+                    // Entry does not exist, lets just create one, even though we're probably not a multimodule ship
+//                    val moduleSlotList = member.variant.moduleSlots
+                    val moduleSlotList = memberVariant.moduleSlots
+                    val variants = if(moduleSlotList == null || moduleSlotList.isEmpty()) {
+                        // It's just this variant
+                        listOf(memberVariant)
+                    } else {
+                        // Otherwise, it's all of them
+                        val mutableVariantList = moduleSlotList
+//                                .map { slot -> member.variant.getModuleVariant(slot) }
+                                .map { slot -> memberVariant.getModuleVariant(slot) }
+                                .toMutableList()
+                        // Obviously, add the 'member' variant to the list as well
+//                        mutableVariantList.add(member.variant)  //TODO get rid of this, we should probably just rename this into childModulesVariantList
+                        mutableVariantList.add(memberVariant)
+
+                        mutableVariantList.toList()
+                    }
+
+                    // Return the optional of the list
+                    Optional.of(variants)
+                }
+
+                val shouldInstallOnMemberVariant = HullmodExoticHandler.shouldInstallHullmodExoticToVariant(
+                        hullmodExotic = hullmodExotic,
+                        parentFleetMember = member,
+                        variant = memberVariant,
+                        variantList = variantListOptional,
+                        workMode = workMode
+                )
+                onShouldCallback.execute(shouldInstallOnMemberVariant, memberVariant)
+                logger.info("onInstall()\tshouldInstallOnMemberVariant: ${shouldInstallOnMemberVariant}, variant: ${memberVariant}")
```

Comment:
1) `log(...)`
2) ~~verbose~~ info is fine.

Reply:
probably done, since there are no explicit calls on `logger` anywhere anymore. resolving.